src/utils/commandExecutor.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2020 Make Positive Provar Ltd
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.md file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as cross_spawn from 'cross-spawn';
+import { fromEvent, interval, Observable, Subscription } from 'rxjs';
+import * as vscode from 'vscode';
+import { CommandBuilder } from './CommandBuilder';
+
+const kill = require('tree-kill');
+
+export class CommandExecutor {
+    public readonly command: CommandBuilder;
+    public readonly cancellationToken?: vscode.CancellationToken;
+    public readonly processExitSubject: Observable<number | undefined>;
+    public readonly processErrorSubject: Observable<Error | undefined>;
+    public readonly stdoutSubject: Observable<Buffer | string>;
+    public readonly stderrSubject: Observable<Buffer | string>;
+
+    private readonly childProcessPid: number;
+
+    constructor(command: CommandBuilder, cancellationToken?: vscode.CancellationToken) {
+        this.command = command;
+        this.cancellationToken = cancellationToken;
+
+        const childProcess = cross_spawn(this.command.command, this.command.args, {});
+
+        this.childProcessPid = childProcess.pid;
+
+        let timerSubscriber: Subscription | null;
+
+        // Process
+        this.processExitSubject = fromEvent(childProcess, 'exit') as Observable<number | undefined>;
+        this.processExitSubject.subscribe((next) => {
+            if (timerSubscriber) {
+                timerSubscriber.unsubscribe();
+            }
+        });
+        this.processErrorSubject = fromEvent(childProcess, 'error') as Observable<Error | undefined>;
```

Comment:
redundant cast as we're not change the type?

Reply:
done